### PR TITLE
Ocaml -> 4.13.1, Add wayland_proxy_virtwl

### PR DESCRIPTION
**Pull request by satmandu**
- This is an alternative to `sommelier`: https://github.com/talex5/wayland-proxy-virtwl
- Wayland proxying works, but X11 proxying is currently not working, [as far as I can tell](https://github.com/talex5/wayland-proxy-virtwl/issues/24).
- Example usage: `stopsommelier ; /usr/local/opam/bin/wayland-proxy-virtwl --tag="test" --wayland-display wayland-1 --x-display=0 --xrdb Xft.dpi:150`  
  - This should create a wayland display at `wayland-1` which can be used by prefixing a wayland command with `WAYLAND_DISPLAY=wayland-1`.
  - e.g. `WAYLAND_DISPLAY=wayland-1 weston-terminal`
  - This requires https://github.com/skycocker/chromebrew/pull/6396 to build
  - Yes the directory structure is wacky, which is why I just pointed it to /usr/local/opam...

Works properly:
- [x] x86_64
- [x] armv7l (Need someone to verify that the binary works... but it builds... if I replace ld with ld.bfd for the build.)
- [ ] i686 (it does build...)

### Run the following to get this pull request's changes locally for testing.
```
CREW_TESTING_REPO=https://github.com/satmandu/chromebrew.git CREW_TESTING_BRANCH=wayland_proxy_virtwl CREW_TESTING=1 crew update
```

### CODE REVIEW

**Comment by uberhacker** [comment]
Unfortunately, I get a segfault on i686.  Does it make sense to create a symlink of the binary targeted in `/usr/local/bin`?  See below:
```
chronos@localhost ~/chromebrew/packages (wayland_proxy_virtwl *=)$ crew files wayland_proxy_virtwl
wayland_proxy_virtwl: Proxy Wayland connections across the VM boundary
/usr/local/opam/bin/wayland-proxy-virtwl
/usr/local/opam/doc/wayland-proxy-virtwl/README.md
/usr/local/opam/lib/wayland-proxy-virtwl/dune-package
/usr/local/opam/lib/wayland-proxy-virtwl/META
/usr/local/opam/lib/wayland-proxy-virtwl/opam
Total found: 5
Disk usage: 1021.47KB
chronos@localhost ~/chromebrew/packages (wayland_proxy_virtwl *=)$ /usr/local/opam/bin/wayland-proxy-virtwl
Segmentation fault (core dumped)
```

**Comment by satmandu** [comment]
That's a good idea. Maybe I should also modify the opam package to remove the self.remove section so we don't clobber installs like this too...

**Comment by uberhacker** [comment]
Would it still work if everything was installed in `/usr/local/share/opam` instead of `/usr/local/opam`?  That would remove the need to bypass FHS compliance.

**Comment by satmandu** [comment]
Yeah that could work. I'd just have to redo the `opam` package...

**Comment by satmandu** [comment]
updated opam path..., but opam needs an i686 rebuild...

File: packages/opam.rb [hunk 1]
```diff
@@ -7,68 +7,63 @@ class Opam < Package
...
   git_hashtag @_ver
 
   binary_url({
-    aarch64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/opam/2.1.1_armv7l/opam-2.1.1-chromeos-armv7l.tpxz',
-     armv7l: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/opam/2.1.1_armv7l/opam-2.1.1-chromeos-armv7l.tpxz',
-       i686: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/opam/2.1.1_i686/opam-2.1.1-chromeos-i686.tar.xz',
-     x86_64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/opam/2.1.1_x86_64/opam-2.1.1-chromeos-x86_64.tpxz'
+    aarch64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/opam/2.1.1-1_armv7l/opam-2.1.1-1-chromeos-armv7l.tpxz',
+     armv7l: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/opam/2.1.1-1_armv7l/opam-2.1.1-1-chromeos-armv7l.tpxz',
+     x86_64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/opam/2.1.1-1_x86_64/opam-2.1.1-1-chromeos-x86_64.tpxz'
```
Comment (uberhacker) [inline packages/opam.rb:18]: Were these binaries generated after the new version of ocaml was installed?

File: packages/wayland_proxy_virtwl.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  @_ver = '5b576d5cad76de6ed7513cc85748efd78ca366cf'
+  version "#{@_ver}-1"
+  compatibility 'all'
+  source_url 'https://github.com/talex5/wayland-proxy-virtwl.git'
+  git_hashtag @_ver
+
+  binary_url({
+    aarch64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/wayland_proxy_virtwl/5b576d5cad76de6ed7513cc85748efd78ca366cf-1_armv7l/wayland_proxy_virtwl-5b576d5cad76de6ed7513cc85748efd78ca366cf-1-chromeos-armv7l.tpxz',
+     armv7l: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/wayland_proxy_virtwl/5b576d5cad76de6ed7513cc85748efd78ca366cf-1_armv7l/wayland_proxy_virtwl-5b576d5cad76de6ed7513cc85748efd78ca366cf-1-chromeos-armv7l.tpxz',
+    x86_64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/wayland_proxy_virtwl/5b576d5cad76de6ed7513cc85748efd78ca366cf-1_x86_64/wayland_proxy_virtwl-5b576d5cad76de6ed7513cc85748efd78ca366cf-1-chromeos-x86_64.tpxz'
```
Comment (uberhacker) [inline packages/wayland_proxy_virtwl.rb:15]: Were these binaries generated after the new versions of ocaml and opam were installed?

File: packages/wayland_proxy_virtwl.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+     armv7l: 'dbc7832b84f64fe292f77cc74e9ee809acb69098072f10d7cab4bf61c3a47070',
+    x86_64: '9d5e0b7ca63b3ec2949f66da4e594741c729a7aa9e488ba16f4a7e6e237d4671'
+  })
+
+  depends_on 'ocaml' => :build
+  depends_on 'opam' => :build
+
+  @OPAMROOT = "#{CREW_PREFIX}/share/opam"
+
+  def self.build; end
```
Comment (uberhacker) [inline packages/wayland_proxy_virtwl.rb:28]: What is the point?

File: packages/wayland_proxy_virtwl.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+  def self.install
+    ENV['OPAMROOT'] = @OPAMROOT
+    system "LINKFLAGS='-fuse-ld=bfd' LDFLAGS='-fuse-ld=bfd' OPAMROOT=#{@OPAMROOT} opam install . --root=#{@OPAMROOT} --destdir=#{CREW_DEST_DIR}#{@OPAMROOT} -y"
```
Comment (uberhacker) [inline packages/wayland_proxy_virtwl.rb:32]: What is the purpose of the period (.) after install?

Comment (satmandu) [reply packages/wayland_proxy_virtwl.rb:32]: Install package from local directory...

**Comment by satmandu** [comment]
FYI this is working w/o acceleration! See https://github.com/talex5/wayland-proxy-virtwl/issues/24

File: packages/wayland_proxy_virtwl.rb (hunk 3 above)
Comment (satmandu) [reply packages/wayland_proxy_virtwl.rb:28]: Point taken...

File: packages/wayland_proxy_virtwl.rb (hunk 2 above)
Comment (satmandu) [reply packages/wayland_proxy_virtwl.rb]: These are going to be replaced... the new ocaml & opam using the new ocaml are a huge PITA...

**Pull request closed without merging**